shared/models/github_provider_credential.rb

```diff
@@ -8,12 +8,13 @@ module FastlaneCI
   class GitHubProviderCredential < ProviderCredential
     attr_accessor :encrypted_api_token
 
-    def initialize(email: nil, full_name: nil, api_token: nil)
+    def initialize(id: nil, email: nil, full_name: nil, api_token: nil, provider_name: nil, type: nil)
```

Comment:
let's revert this change. If you let people provide different `provider_name` and `type` then this is no longer a `GitHubProviderCredential` it becomes more generic. We want to keep this as specific as possible. If we need to allow different `provider_name` and `type` let's create a new `ProviderCredential` subclass to represent that.